drivers/dac/ad5758/ad5758.c

```diff
@@ -102,8 +102,8 @@ static int32_t ad5758_spi_reg_read(struct ad5758_dev *dev,
 		buf[3] = 0x00;
 
 	ret = spi_write_and_read(dev->spi_desc, buf, 4);
-	if (ret < 0)
-		return FAILURE;
+	if (ret)
```

Comment:
The check of ret < 0 is good.
A function may return positive values (maybe not in this case, but in general) which is not an error.

Reply:
In this case, ret = 0 only if the function returns SUCCESS. All functions return SUCCESS or FAILURE.

Reply:
https://github.com/analogdevicesinc/no-OS/blob/1090f222df0ff8c916ebcfe7990098eaa98aee11/include/error.h#L65